llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java

```diff
@@ -247,6 +243,12 @@ public OrcEncodedDataReader(LowLevelCache lowLevelCache, BufferUsageManager buff
     this.jobConf = jobConf;
     // TODO: setFileMetadata could just create schema. Called in two places; clean up later.
     this.evolution = sef.createSchemaEvolution(fileMetadata.getSchema());
+
+    fileIncludes = includes.generateFileIncludes(fileSchema);
```

Comment:
What happens if there are multiple files with different schema?
The test data is created this way:
- Iceberg table created with 2 columns
- Data inserted with 2 columns
- Iceberg table schema modified
- Data inserted with the modified schema
?

Reply:
Ok.. it is covered by the tests.